docs/metrics.md

```diff
@@ -0,0 +1,587 @@
+  subject to this cap.
+
+  :heavy_check_mark: Use `Meter` attributes or `Resource` attributes as
+  appropriate, see [modelling attributes](#modelling-metric-attributes) for
+  details. In the above example, if a process only sells fruits from a
+  particular store, then store_location attribute should be modelled as a
+  Resource - this is not only efficient, but reduced the cardinality capping
+  possibility as well.
+
+// TODO: Document how to pick cardinality limit.
+
+### Memory Preallocation
+
+OpenTelemetry Rust SDK aims to avoid memory allocation on the hot code path.
+When this is combined with [proper use of Metrics API](#metrics-api), heap
+allocation can be avoided on the hot code path.
+
+## Metrics Correlation
+
+Including `TraceId` and `SpanId` as attributes in metrics might seem like an
+intuitive way to achieve correlation with traces or logs. However, this approach
+is ineffective and can make metrics practically unusable. Moreover, it can
+quickly lead to cardinality issues, resulting in metrics being capped.
+
+A better alternative is to use a concept in OpenTelemetry called
+[exemplars](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#exemplar).
+Exemplars provide a mechanism to correlate metrics with traces by sampling
+specific measurements and attaching trace context to them.
+
+Currently, exemplars are not yet implemented in the OpenTelemetry Rust SDK.
+
+## Modelling Metric Attributes
+
+When metrics are being collected, they normally get stored in a [time series
+database](https://en.wikipedia.org/wiki/Time_series_database). From storage and
+consumption perspective, metrics can be multi-dimensional. Taking the [fruit
+example](#example), there are two dimensions - "name" and "color". For basic
+scenarios, all the dimensions can be reported during the [Metrics
+API](#metrics-api) invocation, however, for less trivial scenarios, the
+dimensions can come from different sources:
+
+* [Measurements](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#measurement)
+  reported via the [Metrics API](#metrics-api).
+* Additional attributes provided at meter creation time via
+  [`meter_with_scope`](https://docs.rs/opentelemetry/latest/opentelemetry/metrics/trait.MeterProvider.html#tymethod.meter_with_scope).
+* [Resources](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/sdk.md)
+  configured at the `MeterProvider` level.
+* Additional attributes provided by the collector. For example, [jobs and
+  instances](https://prometheus.io/docs/concepts/jobs_instances/) in Prometheus.
+
+Here is the rule of thumb when modeling the dimensions:
+
+* If the dimension is static throughout the process lifetime (e.g. the name of
+  the machine, data center):
+  * If the dimension applies to all metrics, model it as Resource, or even
```

Comment:
add something about - this attributes are unaffected by cardinality capping and so queried using just these would remain accurate.